kroxylicious-systemtests/src/main/java/io/kroxylicious/systemtests/clients/KafClient.java

```diff
@@ -55,11 +62,50 @@ public void produceMessages(String topicName, String bootstrap, String message,
     }
 
     @Override
-    public String consumeMessages(String topicName, String bootstrap, String messageToCheck, int numOfMessages, Duration timeout) {
+    public List<ConsumerRecord<String, String>> consumeMessages(String topicName, String bootstrap, int numOfMessages, Duration timeout) {
         LOGGER.atInfo().log("Consuming messages using kaf");
         String name = Constants.KAFKA_CONSUMER_CLIENT_LABEL + "-kafka-go";
-        List<String> args = List.of("kaf", "-b", bootstrap, "consume", topicName);
+        List<String> args = List.of("kaf", "-b", bootstrap, "consume", topicName, "--output", "json");
         Job goClientJob = TestClientsJobTemplates.defaultKafkaGoConsumerJob(name, args).build();
-        return KafkaUtils.consumeMessages(topicName, name, deployNamespace, goClientJob, messageToCheck, numOfMessages, timeout);
+        String podName = KafkaUtils.createJob(deployNamespace, name, goClientJob);
+        String log = waitForConsumer(podName, numOfMessages, timeout);
+        LOGGER.atInfo().log(log);
+        List<String> logRecords = List.of(log.split("\n"));
```

Comment:
Don't split the string on newline.. Let ObjectMapper produce you a List<ConsumerRecords> from the log.

Reply:
We have to think about that we don't always have clean logs where to read the JSON and that's all. I needed to go line by line to discard those lines which are not interesting for us.
Sometimes that occurs because a connection failure of the consumer:
```
% Reconnecting to partition [0]: reading offset [0]: end
{"key":"","offset":0,"partition":0,"payload":"Hello-world - 0","timestamp":"2024-06-03T10:54:02.18Z"}
```
I think using `ObjectMapper` does not take into account that kind of issues. That's why I decided to go line by line.